pkg/skaffold/build/tag/git_commit.go

```diff
@@ -57,6 +83,23 @@ func (c *GitCommit) GenerateFullyQualifiedImageName(workingDir string, imageName
 	return fmt.Sprintf("%s:%s", imageName, ref), nil
 }
 
+func (c *GitCommit) makeGitTag(workingDir string) (string, error) {
+	args := make([]string, 0, 4)
+	switch c.variant {
+	case tags:
+		args = append(args, "describe", "--tags", "--always")
+	case commitSha, abbrevCommitSha:
+		args = append(args, "rev-list", "-1", "HEAD")
+		if c.variant == abbrevCommitSha {
+			args = append(args, "--abbrev-commit")
+		}
+	default:
+		return "", fmt.Errorf("this should not happen, please raise an issue")
```

Comment:
this would result in some weird looking error messages:
```Unable to find git commit: this should not happen, please raise an issue```

let's just do something like 
```golang
return "", errors.New("invalid git tag variant %s: defaulting to 'dirty'", c.variant)
```

Reply:
Yeah, sounds much better. However, the `c.variant` is some integer representing an (invalid) variant internally. I don't think it would help to include it in the error message.

Reply:
fair enough!